### DISCUSSION TITLE
[@types/node] Any reasons why OutgoingHttpHeaders is not typed?

### DISCUSSION THREAD

**Discussion by skwee357** [DefinitelyTyped/DefinitelyTyped#66753] [Issues with a @types package]
I wanted to add reply header types to fastify, and created a PR https://github.com/fastify/fastify/pull/5046. However, I was pointed out by one of the contributors that header types actually come from ```@types/node```.

Upon closer inspection, it seems like ```OutgoingHttpHeader``` does not have a concrete type (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/64f8bd1992030d9d85d2717d4fa85e1610be8601/types/node/http.d.ts#L115) and instead typed as ```number | string | string[]```.

```IncomingHttpHeaders```, on the other hand, does have proper typing.

Is there any reason behind it?

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
In TS the type `'accept' | ... | string` is the same as just `string`, isn't it?

**Reply by peterblazejewicz** [reply]
there is a comment in the types, suggesting it was done (long time ago), to allow `number` key type. I don't know api to that plint to verify this.
@Semigradsky wdyt?

**Reply by Semigradsky** [reply]
We can use `number` for http header value, for example for `'Content-Length'` header.

**Reply by Semigradsky** [reply]
@skwee357 I rechecked: the type `type X = 'a' | 'b' | string` is the same as just `string` so there is no profit from proposed changes.

**Reply by skwee357** [reply]
@Semigradsky While numbers for HTTP value is understandable, I'm not sure if it's worth to leave reply header non typed. Request headers are typed, and they can have arbitrary values as well, why can't we work on doing the same for reply headers?

**Reply by Semigradsky** [reply]
@skwee357 ok, I was confused by what you have `OutgoingHttpHeaders` in the question title and `OutgoingHttpHeader` in the question body. 😸 

There is no sense in changing the `OutgoingHttpHeader` type. But yes, we can change `OutgoingHttpHeaders` type to make it similar to `IncomingHttpHeaders` for improving autocomplete.

**Reply by skwee357** [reply]
Added types in a similar manner to ```IncomingHttpHeaders```, see #66783

**Reply by Vampire** [reply]
@skwee357 any reason `content-type` is missing for the outgoing headers?
(⏭#70705)
